src/openvic-simulation/modifier/StaticModifierCache.cpp

```diff
@@ -1,115 +1,166 @@
-	set_static_modifier(hard_player, "hard_player");
-	set_static_modifier(very_hard_player, "very_hard_player");
-	set_static_modifier(very_easy_ai, "very_easy_ai");
-	set_static_modifier(easy_ai, "easy_ai");
-	set_static_modifier(hard_ai, "hard_ai");
-	set_static_modifier(very_hard_ai, "very_hard_ai");
+	set_static_country_modifier(very_easy_player);
+	set_static_country_modifier(easy_player);
+	set_static_country_modifier(hard_player);
+	set_static_country_modifier(very_hard_player);
+	set_static_country_modifier(very_easy_ai);
+	set_static_country_modifier(easy_ai);
+	set_static_country_modifier(hard_ai);
+	set_static_country_modifier(very_hard_ai);
+
+	ret &= add_key_map_entry(
+		key_map, base_modifier.get_identifier(), ONE_EXACTLY,
+		expect_dictionary_keys_and_default(
+			modifier_manager.expect_base_country_modifier(base_modifier),
+			"supply_limit", ZERO_OR_ONE, [this, &modifier_manager](auto const& value) -> bool {
+				return modifier_manager._add_modifier_cb(
+					base_modifier,
+					modifier_manager.get_modifier_effect_cache().get_supply_limit_global_base(),
+					value
+				);
+			}
+		)
+	);
 
-	set_static_modifier(base_modifier, "base_values");
-	set_static_modifier(war, "war");
-	set_static_modifier(peace, "peace");
-	set_static_modifier(disarming, "disarming");
-	set_static_modifier(war_exhaustion, "war_exhaustion");
-	set_static_modifier(infamy, "badboy");
-	set_static_modifier(debt_default_to, "debt_default_to");
-	set_static_modifier(bad_debter, "bad_debter");
-	set_static_modifier(great_power, "great_power");
-	set_static_modifier(secondary_power, "second_power");
-	set_static_modifier(civilised, "civ_nation");
-	set_static_modifier(uncivilised, "unciv_nation");
-	set_static_modifier(literacy, "average_literacy");
-	set_static_modifier(plurality, "plurality");
-	set_static_modifier(generalised_debt_default, "generalised_debt_default");
-	set_static_modifier(total_occupation, "total_occupation", 100);
-	set_static_modifier(total_blockaded, "total_blockaded");
-	set_static_modifier(in_bankruptcy, "in_bankrupcy");
+	set_static_country_modifier(war);
+	set_static_country_modifier(peace);
+	set_static_country_modifier(disarming);
+	set_static_country_modifier(war_exhaustion);
+	set_static_country_modifier(infamy);
+	set_static_country_modifier(debt_default_to);
+	set_static_country_modifier(great_power);
+	set_static_country_modifier(secondary_power);
+	set_static_country_modifier(civilised);
+	set_static_country_modifier(uncivilised);
+	set_static_country_modifier(literacy);
+	set_static_country_modifier(plurality);
+	set_static_country_modifier(generalised_debt_default);
+	set_static_country_modifier(total_occupation);
+	set_static_country_modifier(total_blockaded);
+	
+	constexpr IconModifier::icon_t default_icon = 0;
+	set_country_event_modifier(bad_debtor,"bad_debter", default_icon); //paradox typo
+	set_country_event_modifier(in_bankruptcy,"in_bankrupcy", default_icon); //paradox typo
```

Comment:
Why store these specifically as events? Is there anything that can be done with them that is impossible to do with the others?

Reply:
Yes, these can be checked as if they are normal event modifiers.
Also you if you add an event modifier with the same name, it will override the static modifier.